docs/doc.md

```diff
@@ -134,6 +135,56 @@ That’s our lexer definition it has just token channel which are going to be us
 Syntax analysis reads the stream of `token` which is generated by `lexer`, then make AST(Abstract Synstax Tree) which is passed to compiler.  Koa parser is 'Pratt parser' which is easy to make, modulable and scalable. **The main idea of 'Pratt parser' is each `token` has its own parsing functions.** (infix parsing function, prefix parsing function). In the above diagram `Program` denotes root node of AST and AST consists of slice of `Statements`.
 
 
+### <a name="compile">Compile</a>
+
+Compiling produces a code by assembling information collected from other sources. Koa compiler **reads the `AST` made by `parser` and generates a code called `bytecode`.** `Bytecode` is a kind of assemble codes. This has an information how to execute a program.
+
+#### Bytecode
+
+`Bytecode` is an output generated by compiling. Through executing this code, a machine could execute a program.
+
+```go
+type Bytecode struct {
+   RawByte []byte
+   AsmCode []string
+   Abi     abi.Abi
+}
+```
+
+This is our `Bytecode` structure. It has 3 data. `RawByte` is a byte which has information to execute the program in hexadecimal. `AsmCode` is a collection of assemble codes more readable than byte to human. `Abi` is an interface needed to call the functions by user.
```

Comment:
suggestion!

This is our `Bytecode` structure. It has 3 fields. `RawByte` is a byte which has information of how to execute the program. And `RawByte` consists of hexadecimal code.

Reply:
suggestion!

`AsmCode` is a collection of assemble codes which is more readable to human because this is not bytes but string

Reply:
suggestion!

`Abi` is an interface needed to user for calling the functions.